From b6edf3c86810f80b044751e6fab2c11c9b979ced Mon Sep 17 00:00:00 2001
From: Dan Nica <dan.nica@nxp.com>
Date: Thu, 19 Mar 2020 10:04:44 +0200
Subject: [PATCH 083/269] s32g274a: Enable generic timer

This is required for the mdelay/udelay() family.

Issue: ALB-4274
Upstream-Status: Pending 

Signed-off-by: Dan Nica <dan.nica@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/s32g/platform.mk          |  3 +++
 plat/s32g/s32g274a_bl2_el3.c   |  3 +++
 plat/s32g/s32g274a_bl_common.c | 12 ++++++++++++
 plat/s32g/s32g275_bl31.c       |  5 -----
 4 files changed, 18 insertions(+), 5 deletions(-)
 create mode 100644 plat/s32g/s32g274a_bl_common.c

diff --git a/plat/s32g/platform.mk b/plat/s32g/platform.mk
index ba4fad927..3cdc4f076 100644
--- a/plat/s32g/platform.mk
+++ b/plat/s32g/platform.mk
@@ -25,6 +25,7 @@ PLAT_BL_COMMON_SOURCES	+= plat/s32g/s32g_lowlevel_common.S \
 			   plat/s32g/s32g_linflexuart.S \
 			   plat/s32g/s32g_mc_me.c \
 			   plat/s32g/s32g_ncore.c \
+			   plat/s32g/s32g274a_bl_common.c \
 			   lib/cpus/aarch64/cortex_a53.S
 
 BL2_SOURCES		+= plat/s32g/s32g_lowlevel_bl2.S \
@@ -34,6 +35,8 @@ BL2_SOURCES		+= plat/s32g/s32g_lowlevel_bl2.S \
 			   plat/s32g/s32g274a_storage.c \
 			   plat/s32g/s32g274a_edma.c \
 			   drivers/io/io_storage.c \
+			   drivers/delay_timer/delay_timer.c \
+			   drivers/delay_timer/generic_delay_timer.c \
 			   common/desc_image_load.c \
 			   drivers/nxp/s32g/io/io_memmap.c \
 			   ${DDR_DRV}/ddrss.c \
diff --git a/plat/s32g/s32g274a_bl2_el3.c b/plat/s32g/s32g274a_bl2_el3.c
index 314db5693..7cc392800 100644
--- a/plat/s32g/s32g274a_bl2_el3.c
+++ b/plat/s32g/s32g274a_bl2_el3.c
@@ -18,6 +18,7 @@
 #include "s32g_mc_rgm.h"
 #include "s32g_mc_me.h"
 #include <nxp/s32g/ddr/ddrss.h>
+#include <drivers/generic_delay_timer.h>
 
 static bl_mem_params_node_t s32g_bl2_mem_params_descs[] = {
 	{
@@ -82,6 +83,8 @@ void bl2_el3_early_platform_setup(u_register_t arg0, u_register_t arg1,
 	ncore_init();
 	ncore_caiu_online(A53_CLUSTER0_CAIU);
 
+	generic_delay_timer_init();
+
 	s32g_io_setup();
 }
 
diff --git a/plat/s32g/s32g274a_bl_common.c b/plat/s32g/s32g274a_bl_common.c
new file mode 100644
index 000000000..1bb130b3b
--- /dev/null
+++ b/plat/s32g/s32g274a_bl_common.c
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "platform_def.h"
+
+unsigned int plat_get_syscnt_freq2(void)
+{
+	return COUNTER_FREQUENCY;
+}
diff --git a/plat/s32g/s32g275_bl31.c b/plat/s32g/s32g275_bl31.c
index ea7738c43..104653264 100644
--- a/plat/s32g/s32g275_bl31.c
+++ b/plat/s32g/s32g275_bl31.c
@@ -200,8 +200,3 @@ void bl31_platform_setup(void)
 void bl31_plat_runtime_setup(void)
 {
 }
-
-unsigned int plat_get_syscnt_freq2(void)
-{
-	return COUNTER_FREQUENCY;
-}
-- 
2.17.1

